packages/smol/components/sections/Revoke/AllowancesTable.tsx

```diff
@@ -0,0 +1,164 @@
+import {type ReactElement, useMemo, useState} from 'react';
+import IconChevronPlain from 'packages/lib/icons/IconChevronPlain';
+import {IconSpinner} from 'packages/lib/icons/IconSpinner';
+
+import {AllowanceRow} from './AllowanceRow';
+import {useAllowances} from './useAllowances';
+
+import type {TAddress} from '@builtbymom/web3/types';
+import type {TTokenAllowance} from '@lib/types/Revoke';
+
+type TAllowancesTableProps = {
+	revoke: (tokenToRevoke: TTokenAllowance, spender: TAddress) => void;
+};
+
+type TSortType = {
+	sortBy: 'spender' | 'amount' | 'token' | null;
+	asc: boolean;
+};
+
+export const AllowancesTable = ({revoke}: TAllowancesTableProps): ReactElement => {
+	const {filteredAllowances: allowances, isLoading, isDoneWithInitialFetch} = useAllowances();
+	const isFetchingData = !isDoneWithInitialFetch || isLoading;
+
+	const [sort, set_sort] = useState<TSortType>({sortBy: null, asc: true});
+
+	/**********************************************************************************************
+	 ** Sorting allowances by amount, spender and token. All of them
+	 ** are sorted ether asc or desc order. If sortings are not selected
+	 ** we return allowances in the initial timestamp order
+	 *********************************************************************************************/
+
+	const sortedAllowances = useMemo(() => {
+		if (sort.sortBy === 'amount') {
+			return allowances?.toSorted((a, b) =>
+				sort.asc ? (a.args.value! > b.args.value! ? -1 : 1) : a.args.value! < b.args.value! ? -1 : 1
+			);
+		}
+
+		if (sort.sortBy === 'spender') {
+			return allowances?.toSorted((a, b) =>
+				sort.asc ? b.args.sender.localeCompare(a.args.sender) : a.args.sender.localeCompare(b.args.sender)
+			);
+		}
+
+		if (sort.sortBy === 'token') {
+			return allowances?.toSorted((a, b) =>
+				a.symbol && b.symbol
+					? sort.asc
+						? b.symbol?.localeCompare(a.symbol)
+						: a.symbol?.localeCompare(b.symbol)
+					: 0
+			);
+		}
+
+		return allowances;
+	}, [allowances, sort.asc, sort.sortBy]);
+
+	return (
+		<>
+			{(!allowances || allowances.length === 0) && !isFetchingData ? (
```

Comment:
Just fyi it might be easier to handle conditional rendering with `if return` statements like so -https://github.com/SmolDapp/SmolV2/pull/24/files#diff-3a13eb1c7470e04e8077684c6fed8cfd09fb092a4a4df935d7638e3540328501R118